client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
```

Comment:
> I wonder if this could be an argument against keeping them alongside the action creators

as @artpi mentioned, this is the main difference between our API middleware PRs. I do not think that the reader should care or know anything about notices. why should it? the reader knows that it needs to tell Calypso to delete a post or something like that - end of story. it's now up to Calypso to run that action and respond accordingly. if it should fail, then that failure can and will trigger a notice. what if we're offline? do we need to make the reader know about online/offline status?

if the logic for actually accomplishing the action is spread out we will end up with hard to manage spaghetti, but if we split it out then we can take advantage of code reuse, simplification of the independent functional blocks, and management of global/central policies.